### PR TITLE
exit when any command fails

### PR DESCRIPTION
**Pull request by scytacki**
and then additionally this prints out information about the failing command
and what its exit code was
Taken from here: https://intoli.com/blog/exit-on-errors-in-bash-scripts/

### CODE REVIEW

**Review by dstrawberrygirl — CHANGES_REQUESTED** [review]
While these changes work fine when a build is correctly configured, I just tested with invalid s3 credentials on Seismic Explorer (to reproduce the 403 failures we saw previously). The errors were swallowed, and all that was displayed was the following:

```
[info] Deploying _site/* to models-resources
[info] Summary: There was nothing to push
```
I'm not sure what to suggest as a change here since I'm not an expert on sh syntax, but I can test fairly easily if you have ideas on what to change

**Comment by scytacki** [comment]
@dstrawberrygirl Are you sure that it actually tried to push something to S3 and failed? 
I think the S3 bucket is open to the public for reading. So s3_website can probably look at the current files in the bucket and decide not to push anything without running into a 403.

Can you try changing a file, so s3_website has something new to push?

File: s3_deploy.sh [hunk 1]
```diff
@@ -1,5 +1,13 @@
 #!/bin/bash
 
+# exit when any command fails
+set -e
+
+# keep track of the last executed command
+trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
+# echo an error message before exiting
+trap 'echo "\"${last_command}\" command filed with exit code $?."' EXIT
```
Comment (scytacki) [inline s3_deploy.sh:9]:
```suggestion
trap 'echo "\"${last_command}\" command failed with exit code $?."' EXIT
```

**Comment by scytacki** [comment]
@dstrawberrygirl also based on this: 
https://github.com/concord-consortium/seismic-explorer/runs/1414973246#step:7:29
It seems like the aws invalidation command completed successfully. 

How did you make the credentials invalid? Was it by changing the secrets in GH, or did you make a change in the AWS policy for the associated user?

All symptoms point to the credentials still be valid some how.

**Comment by dstrawberrygirl** [comment]
@scytacki I replaced the credentials with the incorrectly-named user credentials to reproduce the error from earlier in the day, but it looks like the s3 deployment was not done because the js files had not been changed. I just tested with updated application logic and that did trigger the push, which failed as expected!

I fixed the credentials and did a re-run for the job and it passed as expected. This looks good to me! 👍

**Review by dstrawberrygirl — APPROVED** [review]

**Pull request merged**